pygmt/clib/session.py

```diff
@@ -856,8 +860,8 @@ def put_matrix(self, dataset, matrix, pad=0):
         The dataset must be created by :meth:`pygmt.clib.Session.create_data`
         first. Use ``|GMT_VIA_MATRIX'`` in the family.
 
-        Not at all numpy dtypes are supported, only: float64, float32, int64,
-        int32, uint64, and uint32.
+        Not at all numpy dtypes are supported, only: int8, int16, int32, int64,
+        uint8, uint16, uint32, uint64, float32, float64, str\_ and datetime64.
```

Comment:
I just realized that passing `str_` type matrix to GMT is allowed in the PyGMT `put_matrix` function, but was not documented here. But I'm wondering what happens if passing a `str` matrix to GMT.